server/plugin/plugin.go

```diff
@@ -376,14 +379,59 @@ func (p *Plugin) CreateBotDMPost(userID, message, postType string) {
 	}
 }
 
+func (p *Plugin) CheckISDuplicateSummary(userName, text string) (bool, error) {
+	p.API.LogWarn("checking CheckISDuplicateSummary", "userName", userName)
+	previousSummary, err := p.GetDailySummaryText(userName)
+	p.API.LogWarn("checking previousSummary", "previousSummary", previousSummary)
+	if err != nil {
+		return false, err
+	}
+	if previousSummary == text {
+		return true, nil
+	}
+	return false, nil
+}
+func (p *Plugin) StoreDailySummaryText(userName, summaryText string) (bool, error) {
+	if err := p.API.KVSet(userName+dailySummary, []byte(summaryText)); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+func (p *Plugin) GetDailySummaryText(userName string) (string, error) {
+	summaryByte, err := p.API.KVGet(userName + dailySummary)
+	if err != nil {
+		return "", err
+	}
+	if summaryByte == nil {
+		return "", nil
+	}
+	p.API.LogWarn("GetDailySummaryText", "summary text ", string(summaryByte))
+	return string(summaryByte), nil
+}
+
 func (p *Plugin) PostToDo(info *GitHubUserInfo) {
```

Comment:
0/5 Maybe this should return an `error` and we let the caller deal with the logging.

Reply:
it returns err and text value which got stored

Reply:
I think hanzei is saying that `PostToDo` should return an `error`. Then the caller of `PostToDo` will log the error, to avoid several `LogWarn` calls in the same function.

Reply:
that part of code was an existing code , should i change as you are asking for

Reply:
done